tests/tests.sh

```diff
@@ -19,73 +20,84 @@ function print_services_logs {
     docker-compose logs postfix
-    echo "postgresql logs:"
+    echo "Postgresql logs:"
     docker-compose logs postgres
-    echo "nginx logs:"
+    echo "Nginx logs:"
     docker-compose logs nginx
-    echo "nginx logs:"
+    echo "Redis logs:"
     docker-compose logs redis
 }
 
-function admin_login {
+function test_admin_login {
     # This function is used to login into django-admin
     # It creates the cookie.txt file that containes CSRF
     # token and session ID.
-    $CURL_BIN $LOGIN_URL > /dev/null
+    $CURL_BIN ${APP_URL}/admin/login/ > /dev/null
     DJANGO_TOKEN="csrfmiddlewaretoken=$(grep csrftoken $COOKIES | \
                     sed 's/^.*csrftoken[[:blank:]]*//')"
     $CURL_BIN -d "$DJANGO_TOKEN;username=$USERNAME;password=$PASSWORD" \
-                -X POST --dump-header - $LOGIN_URL | grep -q "302 Found" && \
+                -X POST --dump-header - ${APP_URL}/admin/login/ | grep -q "302 Found" && \
                 { echo "SUCCESS: Login request acknowledgement received!"; } || \
                 { echo "ERROR: Login request acknowledgement not received!"; FAILURE=1; }
 }
 
 function test_dashboard_login {
     # This tests the login was successful; being able to
     # access this page also means redis is running.
-    $CURL_BIN --head http://dashboard.openwisp.org/admin/ | grep -q "200 OK" && \
-              { echo "SUCCESS: Admin login successful!"; } || \
-              { echo "ERROR: Admin login failed!"; FAILURE=1; }
+    $CURL_BIN --head ${APP_URL}/admin/ | grep -q "200 OK" && \
+                { echo "SUCCESS: Admin login successful!"; } || \
+                { echo "ERROR: Admin login failed!"; FAILURE=1; }
 }
 
 function test_postfix {
     # This test ensures that postfix service is running
     # correctly by making a request to reset password.
-    $CURL_BIN $LOGIN_URL > /dev/null
+    $CURL_BIN ${APP_URL}/admin/login/ > /dev/null
     DJANGO_TOKEN="csrfmiddlewaretoken=$(grep csrftoken $COOKIES | \
                   sed 's/^.*csrftoken[[:blank:]]*//')"
     $CURL_BIN -d "$DJANGO_TOKEN;email=admin@example.com" \
               -X POST --dump-header - \
-              http://dashboard.openwisp.org/accounts/password/reset/ | grep -q "302 Found" && \
+              ${APP_URL}/accounts/password/reset/ | grep -q "302 Found" && \
             { echo "SUCCESS: Postfix service is working!"; } || \
             { echo "ERROR: Postfix service did not respond!"; FAILURE=1; }
 }
 
 function test_websocket {
     # This test ensures that wesocket service is running
     # correctly by trying to reach \ws location in nginx.
-    timeout 2 $CURL_BIN --include \
+    $CURL_BIN -I -X GET --include \
                         --no-buffer \
                         --header "Connection: Upgrade" \
                         --header "Upgrade: websocket" \
                         --header "Host: dashboard.openwisp.org" \
                         --header "Origin: https://dashboard.openwisp.org" \
```

Comment:
Looks like you removed most of the hard coding of 'dashboard.openwisp.org' except for here.  Shouldn't matter for controlled tests but it seems inconsistent.

Reply:
:+1:

Reply:
I am not too worried about this file because the shell script tests have failed me once or twice now, so I am moving tests to selenium. 
However, I have fixed this for now! :smile: